cordformation/src/main/kotlin/net/corda/plugins/DockerImage.kt

```diff
@@ -0,0 +1,100 @@
+package net.corda.plugins
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import com.spotify.docker.client.DefaultDockerClient
+import org.apache.commons.lang.StringUtils
+import java.nio.file.Path
+import java.nio.file.Paths
+
+
+open class DockerImage : DefaultTask() {
+
+    private companion object{
+        //Currently a bug in gradle's projectFile that writes the file with a lower case 'd'
+        //using this name ensures that the docker plugin can find the file as expected.
+        private const val DOCKER_FILE_NAME = "dockerfile"
+    }
+
+    init {
+        description = "Creates a docker file and immediately builds that image to the local repository."
+        group = "CordaDockerDeployment"
+    }
+
+    @get:Optional
+    @get:Input
+    internal var baseImage: String? = null
+
+    fun baseImage(baseImage: String) {
+        this.baseImage = baseImage
+    }
+
+    @get:Optional
+    @get:Input
+    internal var trustRootStoreName: String? = null
+
+    fun trustRootStoreName(trustRoot: String) {
+        this.trustRootStoreName = trustRoot
+    }
+
+
+    @TaskAction
+    fun build() {
+
+        logger.lifecycle("Running DockerImage task")
+        project.copy {
```

Comment:
Ah i had heard rumours there were nicer ways of doing it but hadn't found out where they were. 
Will do

Reply:
There is another version of this coming, probably in the morning, 
as it now needs to collect the workflows jar, the contracts jar AND any cordapp dependency jars e.g tokens
Will do my best to get all the fixes in and get your opinion on the new version tomorrow. 

Thank you for all the time you are putting into the feedback to by the way 💯

Reply:
> it now needs to collect the workflows jar, the contracts jar AND any cordapp dependency jars e.g tokens

This is where the `@InputFiles FileCollection` comes in - the task exposes one of those and _the users put whatever they like_ into it from their `build.gradle` file. All the task has to do is iterate over its contents.

Also note that if you add a _task_ to a `FileCollection` then not only will Gradle put the outputs of that task into your collection, it will also _automatically_ create a dependency between that task and yours.